src/catalog/trigger_catalog.cpp

```diff
@@ -0,0 +1,302 @@
+      {"trigger_name", "database_oid", "table_oid"}, TRIGGER_CATALOG_NAME "_skey2",
+      false, IndexType::BWTREE, txn);
+
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"trigger_oid"}, TRIGGER_CATALOG_NAME "_skey3",
+      false, IndexType::BWTREE, txn);  
+}
+
+TriggerCatalog::~TriggerCatalog() {}
+
+bool TriggerCatalog::InsertTrigger(
+    std::string trigger_name, oid_t database_oid, oid_t table_oid,
+    int16_t trigger_type,
+    std::string fire_condition, //TODO: this actually should be expression
+    std::string function_name,
+    std::string function_arguments,
+    int64_t time_stamp, type::AbstractPool *pool,
+    concurrency::Transaction *txn) {
+
+
+  std::unique_ptr<storage::Tuple> tuple(
+      new storage::Tuple(catalog_table_->GetSchema(), true));
+
+
+  LOG_INFO("type of trigger inserted:%d", trigger_type);
+
+  auto val0 = type::ValueFactory::GetIntegerValue(oid_++ | TRIGGER_OID_MASK);
+  auto val1 = type::ValueFactory::GetVarcharValue(trigger_name, pool);
+  auto val2 = type::ValueFactory::GetIntegerValue(database_oid);
+  auto val3 = type::ValueFactory::GetIntegerValue(table_oid);
+  auto val4 = type::ValueFactory::GetIntegerValue(trigger_type);
+  auto val5 = type::ValueFactory::GetVarcharValue(fire_condition, pool);
+  auto val6 = type::ValueFactory::GetVarcharValue(function_name, pool);
+  auto val7 = type::ValueFactory::GetVarcharValue(function_arguments, pool);
+  auto val8 = type::ValueFactory::GetIntegerValue(time_stamp);
+
+  tuple->SetValue(ColumnId::TRIGGER_OID, val0, pool);
+  tuple->SetValue(ColumnId::TRIGGER_NAME, val1, pool);
+  tuple->SetValue(ColumnId::DATABASE_OID, val2, pool);
+  tuple->SetValue(ColumnId::TABLE_OID, val3, pool);
+  tuple->SetValue(ColumnId::TRIGGER_TYPE, val4, pool);
+  tuple->SetValue(ColumnId::FIRE_CONDITION, val5, pool);
+  tuple->SetValue(ColumnId::FUNCTION_NAME, val6, pool);
+  tuple->SetValue(ColumnId::FUNCTION_ARGS, val7, pool);
+  tuple->SetValue(ColumnId::TIME_STAMP, val8, pool);
+
+  // Insert the tuple
+  return InsertTuple(std::move(tuple), txn);
+}
+
+ResultType TriggerCatalog::DropTrigger(const std::string &database_name,
+                        const std::string &table_name,
+                       const std::string &trigger_name,
+                       concurrency::Transaction *txn) {
+
+  if (txn == nullptr) {
+    LOG_TRACE("Do not have transaction to drop trigger: %s", table_name.c_str());
+    return ResultType::FAILURE;
+  }
+
+  // Checking if statement is valid
+  oid_t database_oid =
+    DatabaseCatalog::GetInstance()->GetDatabaseOid(database_name, txn);
+  if (database_oid == INVALID_OID) {
+    LOG_TRACE("Cannot find database  %s!", database_name.c_str());
+    return ResultType::FAILURE;
+  }
+
+  oid_t table_oid =
+    TableCatalog::GetInstance()->GetTableOid(table_name, database_oid, txn);
+  if (table_oid == INVALID_OID) {
+    LOG_TRACE("Cannot find table %s!", table_name.c_str());
```

Comment:
Thanks for pointing it out. I don't know much about the difference between DEBUG and TRACE before. After learning from https://softwareengineering.stackexchange.com/questions/279690/why-does-the-trace-level-exist-and-when-should-i-use-it-rather-than-debug, I find the current ways is interestingly correct. Trace is used for finer-gain debug. So in each small step, it uses TRACE, while at the end, it uses DEBUG. I am really glad that you point it out.